oap-server/server-configuration/grpc-configuration-sync/src/main/test/java/org/apache/skywalking/oap/server/configuration/grpc/GRPCConfigurationTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.grpc;
+
+import io.grpc.testing.GrpcServerRule;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.configuration.api.ConfigChangeWatcher;
+import org.apache.skywalking.oap.server.configuration.api.GroupConfigChangeWatcher;
+import org.apache.skywalking.oap.server.configuration.service.ConfigurationServiceGrpc;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@Slf4j
+public class GRPCConfigurationTest {
+    private GRPCConfigurationProvider provider;
+    private GRPCConfigWatcherRegister register;
+    private ConfigChangeWatcher singleWatcher;
+    private GroupConfigChangeWatcher groupWatcher;
+    /**
+     * 0:init, 1:change, 2:no change, 3:delete
+     */
+    public static volatile int singleDataFlag;
+    /**
+     * 0:init, 1:change, 2:no change, 3:delete
+     */
+    public static volatile int groupDataFlag;
```

Comment:
Using a static variable to pass states between classes is not a good practice, although it's only a test, please consider other mechanism, e.g. passing an `AtomicInteger` as a constructor parameter of `MockGRPCConfigService` class